### PR TITLE
add azure_rm_rediscache reboot/regenerate key

### PR DESCRIPTION
**Pull request by yungezz**
##### SUMMARY

support reboot/regenerate  key in azure_rm_rediscache

##### ISSUE TYPE

- Feature Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/modules/cloud/azure/azure_rm_rediscache.py [hunk 1]
```diff
@@ -109,6 +109,28 @@
...
+                default: all
+    regenerate_key:
+        description:
+            - Regenerate Redis cache's access keys.
+        suboptions:
+            key_type:
+                description: The Redis key to regenerate.
+                choices:
+                    - primary
+                    - Secondary
```
Comment (zikalino) [inline lib/ansible/modules/cloud/azure/azure_rm_rediscache.py]: Secondary is uppercase

**Review by zikalino — APPROVED** [review]
Just fix "Secondary" in the doc, and good to go

File: lib/ansible/modules/cloud/azure/azure_rm_rediscache.py [hunk 2]
```diff
@@ -109,6 +109,28 @@
...
+                description: If clustering is enabled, the id of the shard to be rebooted.
+                type: int
+            reboot_type:
+                description: Which Redis node(s) to reboot.
+                choices:
+                    - primary
+                    - secondary
+                    - all
+                default: all
+    regenerate_key:
```
Comment (zikalino) [inline lib/ansible/modules/cloud/azure/azure_rm_rediscache.py:125]: here's just a suggestion:
maybe we could have 2 options "regenerate_primary_key", and "regenerate_secondary_key" Boolean options?

Comment (yungezz) [reply lib/ansible/modules/cloud/azure/azure_rm_rediscache.py:125]: do you think Matt will like the idea :)

**Pull request merged**